### PR TITLE
Bug/fix enum options

### PR DESCRIPTION
**Pull request by swebb**
Fix a bug where enum was using strings instead of symbols.

Also explicitly set enum values (I feel this is clearer but its just my preference).

### CODE REVIEW

**Comment by jordanmaguire** [comment]
this should be getting merged into release branch

File: lib/model_configuration/attribute.rb [hunk 1]
```diff
@@ -45,9 +45,14 @@ def type
     def as_enum
       # Should look like:
       #   enum attribute_name: ["one", "two"]
+
       if is_enum?
-        enum_options_as_string = properties[:enum_options].collect {|x| "\"#{x}\""}.join(", ")
-        "enum #{name}: [#{enum_options_as_string}]"
+        if (enum_options = properties[:enum_options]).present?
+          enum_options_as_hash = enum_options.each_with_index.collect {|key, index| "#{key}: #{index}"}.join(", ")
```
Comment (jordanmaguire) [inline lib/model_configuration/attribute.rb]: Use `Frontier::HashDecorator`

File: spec/lib/model_configuration/attribute/factory_declaration_spec.rb [hunk 2]
```diff
@@ -35,18 +35,10 @@
...
-      context "when enum_options is set" do
-        let(:enum_options) { ["one", "two"] }
-        it { should eq("field_name { [\"one\", \"two\"].sample }") }
-      end
-
-      context "when enum_options is not set" do
-        let(:enum_options) { nil }
-        specify { expect { subject }.to raise_error(ArgumentError) }
-      end
+      it { should eq("field_name { TestModel.field_names.keys }") }
```
Comment (jordanmaguire) [inline spec/lib/model_configuration/attribute/factory_declaration_spec.rb]: Should have a `.sample` on the end.

File: lib/model_configuration/attribute.rb [hunk 3]
```diff
@@ -45,9 +45,14 @@ def type
...
+          "enum #{name}: {#{enum_options_as_hash}}"
+        else
+          raise(ArgumentError, "No enum_options provided for attribute: #{name}")
+        end
```
Comment (jordanmaguire) [inline lib/model_configuration/attribute.rb]: Why is a function called `is_enum?` returning this?

Comment (jordanmaguire) [inline lib/model_configuration/attribute.rb]: Oh right - I see.

**Comment by jordanmaguire** [comment]
:+1: othr than the merge target.

**Comment by swebb** [comment]
Going to manually merge this into the release branch.

**Pull request closed without merging**